src/task/create-item-validation-task.ts

```diff
@@ -0,0 +1,117 @@
+// global
+import { DatabaseTransactionHandler, Item, ItemService, Member } from 'graasp';
+// local
+import { ItemValidationService } from '../db-service';
+import { BaseValidationTask } from './base-validation-task';
+import {
+  ItemValidationProcesses,
+  ItemValidationReviewStatuses,
+  ItemValidationStatuses,
+  ITEM_TYPE,
+} from '../constants';
+import { getStatusIdByName, stripHtml } from '../utils';
+import { checkBadWords } from '../processes/badWordsDetection';
+
+type InputType = { itemId: string };
+
+export class CreateItemValidationTask extends BaseValidationTask<string> {
+  input: InputType;
+  itemService: ItemService;
+
+  get name(): string {
+    return CreateItemValidationTask.name;
+  }
+
+  constructor(
+    member: Member,
+    validationService: ItemValidationService,
+    itemService: ItemService,
+    input: InputType,
+  ) {
+    super(member, validationService);
+    this.itemService = itemService;
+    this.input = input;
+  }
+
+  async run(handler: DatabaseTransactionHandler): Promise<void> {
+    this.status = 'RUNNING';
+
+    const { itemId } = this.input;
+
+    // create record in item-validation
+    const iVId = await this.validationService.createItemValidation(
+      itemId,
+      handler,
+    );
+    // variable to indicate if manual review is needed
+    let needReview = false;
+
+    // get all enabled processes
+    const enabledProcesses = await this.validationService.getEnabledProcesses(handler);
+
+    // get map for all statuses
+    const iVStatuses = await this.validationService.getItemValidationStatuses(handler);
+
+    // get item
+    const item = await this.itemService.get(itemId, handler);
+
+    // recursively validate item and subitems
+    const validateItem = async (item: Item) => {
+      // run each process on item
+      enabledProcesses.forEach(async (process) => {
+        // create pending entry
+        const iVGEntry = await this.validationService.createItemValidationGroup(
+          item?.id,
+          iVId,
+          process.id,
+          getStatusIdByName(iVStatuses, ItemValidationStatuses.Pending),
+          handler,
+        );
+
+        let status = '';
```

Comment:
What if status stays empty?? I think you should throw an error if the process wasn't found (default case of the switch) or/and set status to error ?